aporia.nim

```diff
@@ -125,11 +125,46 @@ proc updateTabUI(t: Tab) =
   else:
     t.label.setText(name)
   setTabTooltip(t)
+ 
+proc checkFileUpdate(pageNum : int): int = 
+  ## Checks if the current tab open has been updated outside of Aporia.
+  ## Returns ``1`` if the file has been updated, ``0`` if not, and ``-1``
+  ## if there has been no change
+
+  var changedInfo: FileInfo = getFileInfo(win.tabs[pageNum].filename)
+  result = -1
+  if win.tabs[pageNum].lastEdit != changedInfo.lastWriteTime:
+
+    var filename: string = extractFilename(win.tabs[pageNum].filename)
+    var askUpdate = win.w.messageDialogNew(0, MessageWarning, BUTTONS_NONE, nil)
```

Comment:
Please use the same logic to show a message as when a file cannot be opened. This may require some refactoring unfortunately so if you don't want to do it then that's fine. Message boxes are in general annoying and I like to avoid them.

Here is where what I am talking about is used: https://github.com/nim-lang/Aporia/blob/master/aporia.nim#L828